sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -115,6 +116,19 @@ case class InsertIntoHiveTable(
     }
   }
 
+  def shuffleByDynamicCol(
+      rdd: RDD[InternalRow],
+      writerContainer: SparkHiveWriterContainer): RDD[InternalRow] = writerContainer match {
```

Comment:
I'd prefer moving this pattern matching out of this method and put it in `InsertIntoHiveTable`.